packages/auto-approve/src/language-rules/java.ts

```diff
@@ -0,0 +1,117 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {File} from '../get-pr-info';
+import {logger} from 'gcf-utils';
+import {
+  getJavaVersions,
+  FileSpecificRule,
+  runVersioningValidation,
+  isOneDependencyChanged,
+  Versions,
+  doesDependencyChangeMatchPRTitleJava,
+} from '../utils-for-pr-checking';
+
+export const PERMITTED_FILES = [
+  {
+    prAuthor: 'renovate-bot',
+    process: 'dependency',
+    targetFile: /pom.xml$/,
+    // This would match: chore(deps): update dependency com.google.cloud:google-cloud-datacatalog to v1.4.2 or chore(deps): update dependency com.google.apis:google-api-services-policytroubleshooter to v1-rev20210319-1.32.1
+    title: /^(fix|chore)\(deps\): update dependency (@?\S*) to v(\S*)$/,
+    /* This would match:
+          <groupId>com.google.apis</groupId>
+          <artifactId>google-api-services-policytroubleshooter</artifactId>
+          -      <version>v1-rev20210319-1.31.5</version>
+          or
+          <groupId>com.google.apis</groupId>
+          <artifactId>google-api-services-policytroubleshooter</artifactId>
+    -     <version>v1-rev20210319-1.31.5</version>
+        */
+    oldVersion:
+      /<groupId>([^<]*)<\/groupId>[\s]*<artifactId>([^<]*)<\/artifactId>[\s]*-[\s]*<version>(v[0-9]-rev[0-9]*-([0-9]*)\.([0-9]*\.[0-9])|([0-9]*)\.([0-9]*\.[0-9]*))<\/version>[\s]*/,
+    /* This would match:
+          <groupId>com.google.cloud</groupId>
+          <artifactId>google-cloud-datacatalog</artifactId>
+    -     <version>1.4.1</version>
+    +     <version>1.4.2</version>
+          or
+           <groupId>com.google.apis</groupId>
+           <artifactId>google-api-services-policytroubleshooter</artifactId>
+    -      <version>v1-rev20210319-1.31.5</version>
+    +      <version>v1-rev20210319-1.32.1</version>
+        */
+    newVersion:
+      /<groupId>([^<]*)<\/groupId>[\s]*<artifactId>([^<]*)<\/artifactId>[\s]*-[\s]*<version>(v[0-9]-rev[0-9]*-[0-9]*\.[0-9]*\.[0-9]|[[0-9]*\.[0-9]*\.[0-9]*)<\/version>[\s]*\+[\s]*<version>(v[0-9]-rev[0-9]*-([0-9]*)\.([0-9]*\.[0-9])|([0-9]*)\.([0-9]*\.[0-9]*))<\/version>/,
+  },
+];
+
+export class Rules {
```

Comment:
Rather than each language creating a `Rules` class. We should have a base `Rules` interface with something like:

```
interface Rules {
  checkPR(): Promise<boolean>;
}
```

Each language would implement the `Rules` interface. Then the calling code wouldn't care about any implementation details of where the rules came from -- all language rules would be treated the same way.